backend/controllers/myFavoritePostController.js

```diff
@@ -0,0 +1,78 @@
+const asyncHandler = require('express-async-handler');
+const Post = require('../models/postModel');
+const User = require('../models/userModel');
+const MyFavoritePost = require('../models/myFavoritePostModel');
+const ObjectId = require('mongodb').ObjectId;
+
+// @desc    Get all my favorite posts
+// @route   GET /api/myFavoritePosts
+// @access  Public
+const getAllMyFavoritePosts = asyncHandler(async (req, res) => {
+
+  const pipeline = [
+    {
+      '$match': {
+        'user': ObjectId(req.user.id)
+      }
+    }, {
+      '$lookup': {
+        'from': 'posts', 
+        'localField': 'post', 
+        'foreignField': '_id', 
+        'as': 'post'
+      }
+    }, {
+      '$sort': {
+        'updatedAt': -1
+      }
+    }
+  ];
+  const myFavoritePosts = await MyFavoritePost.aggregate(pipeline);
```

Comment:
Data interfacing should be separate from response/request handling. This logic should be in the model or data interface, in the pattern we've used for all our in-class projects.

Reply:
Same comment applies throughout these Controller files.

Reply:
Hi Cheri,
The controller files here play the same roles as data interface files that we learn in the class, and all the files in routes folder work as the same routes files from our class projects. I think it's just the naming issue.
Best,
Joseph